gst/nnstreamer/tensor_source/tensor_src_iio.c

```diff
@@ -1008,15 +1229,83 @@ gst_tensor_src_iio_start (GstBaseSrc * src)
     }
   }
 
+  /** set source pad */
+  GstCaps *prev_caps, *caps, *updated_caps;
+  GstPad *pad;
+
+  self->srcpad = src->srcpad;
+  pad = self->srcpad;
+  gst_pad_use_fixed_caps (pad);
+
+  /**
+   * We can update the caps of the pad in current NULL state
+   * as the pad is activated in READY->PAUSED trainsition.
+   */
+  if (!gst_tensor_src_iio_create_config (self)) {
+    GST_ERROR_OBJECT (self, "Error creating config.\n");
+    goto error_channels_free;
+  }
+
+  if (self->is_tensor) {
+    GstTensorConfig tensor_config;
+    gst_tensor_info_copy (&tensor_config.info,
+        &(self->tensors_config->info.info[0]));
+    tensor_config.rate_n = self->tensors_config->rate_n;
+    tensor_config.rate_d = self->tensors_config->rate_d;
+    caps = gst_tensor_caps_from_config (&tensor_config);
+  } else {
+    caps = gst_tensors_caps_from_config (self->tensors_config);
+  }
+  if (caps == NULL) {
+    GST_ERROR_OBJECT (self, "Error creating caps from config.\n");
+    goto error_config_free;
+  }
+
+  /** get current caps, take intersection with it and then set the new ones */
+  if (gst_pad_has_current_caps (pad)) {
+    prev_caps = gst_pad_get_current_caps (pad);
+    if (gst_caps_can_intersect (caps, prev_caps)) {
+      updated_caps = gst_caps_intersect (caps, prev_caps);
+    } else {
+      GST_ERROR_OBJECT (self,
+          "No intersection between new and current caps.\n");
+      goto error_caps_free;
+    }
+    gst_caps_unref (caps);
+    gst_caps_unref (prev_caps);
+  } else {
+    updated_caps = caps;
+  }
+
+  if (!gst_pad_set_caps (pad, updated_caps)) {
+    GST_ERROR_OBJECT (self, "Unable to set caps for the pad.\n");
+    gst_caps_unref (updated_caps);
+    goto error_config_free;
+    /** free this in stop as well */
+  }
+
+  if (DBG) {
+    GST_DEBUG_OBJECT (self, "Finalization of caps while starting device");
+    GST_DEBUG_OBJECT (self, "caps = %" GST_PTR_FORMAT, updated_caps);
+  }
+
+  gst_caps_unref (updated_caps);
 
 safe_return:
   self->configured = TRUE;
-  /** set the source as live */
-  gst_base_src_set_live (src, TRUE);
+  /** bytes every buffer will be fixed */
+  gst_base_src_set_dynamic_size (src, FALSE);
   /** complete the start of the base src */
   gst_base_src_start_complete (src, GST_FLOW_OK);
   return TRUE;
 
+error_caps_free:
+  gst_caps_unref (prev_caps);
+  gst_caps_unref (caps);
+
+error_config_free:
+  g_free (self->tensors_config);
```

Comment:
In gst_tensor_src_iio_create_config()
```
gst_tensor_info_copy (&config->info.info[info_idx], &info[info_idx]);
...
tensor_src_iio->tensors_config = config;
```

before free tensors_config, it needs gst_tensors_info_free() (name newly allocated)

Reply:
Done.